index.js

```diff
@@ -100,23 +75,19 @@ app.get('/category/:ctyId', (req, res) => {
     categoryId = `cty${userInput.padStart(2, '0')}`
   }
 
-  // Getting category name
-  for (let category of categoryData.categories) {
-    if (category.id === categoryId) {
-      categoryNameVal = category.categoryName;
-      break;
-    }
-  }
+  // Creating required Maps
+  dataService.getCategories;
+  dataService.getCombinedProductMap
 
-  const jsonOutputObject = {
-    category: categoryId,
-    categoryName: categoryNameVal,
+  jsonOutputObject = {
+    ctyId: categoryId,
+    categoryName: dataService.categoryMap[categoryId],
     products: []
   };
 
-  for (let product of productData.products) {
-    if (product.categoryId === categoryId) {
-      jsonOutputObject.products.push(product)
+  for(product of dataService.combinedProductMap) {
```

Comment:
Use `getCombinedProductMap` instead of this.
```suggestion
  for(product of dataService.getCombinedProductMap) {
```

Reply:
Done